Dockerfile

```
@@ -1,22 +1,29 @@
-FROM pindar/jq as jq
+# === Build builder image ===
 
-FROM gradle:6.8.3-jdk11 
+FROM gradle:6.8.3-jdk11 AS build
 
-COPY --from=jq /usr/local/bin/jq /usr/local/bin/jq
-RUN chmod +x /usr/local/bin/jq
+WORKDIR /home/builder
 
-COPY lib/build.gradle /opt/test-runner/lib/build.gradle
+# Prepare required project files
+COPY lib/src ./src
+COPY build.gradle ./
```

I got the following error when building:

> => ERROR [build 4/5] COPY build.gradle ./ 

This was due to this line:

```suggestion
COPY build.gradle ./
```

I tried to fix this by doing:

```suggestion
COPY lib/build.gradle ./
```

This then causes the error to go away, but I get the next error:

```
#10 16.74 An exception occurred applying plugin request [id: 'com.github.johnrengelman.shadow', version: '7.0.0']
#10 16.74 > Failed to apply plugin class 'com.github.jengelman.gradle.plugins.shadow.ShadowBasePlugin'.
#10 16.74    > This version of Shadow supports Gradle 7.0+ only. Please upgrade.
```

It looks like the `com.github.johnrengelman.shadow` plugin required Gradle 7+ but 6.8.3 is used in the Dockerfile.

---

Ah yes, I see the issue there is that I need to switch shadow to version 6.1.0 for compatibility with Gradle 6 (per https://github.com/johnrengelman/shadow). It was using Gradle 7+ on the Gitpod system where I was testing this.

---

👍